graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/remote/SearchJobsStatusResource.java

```diff
@@ -0,0 +1,83 @@
+ */
+package org.graylog.plugins.views.search.rest.remote;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.container.AsyncResponse;
+import jakarta.ws.rs.container.Suspended;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import okhttp3.ResponseBody;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.graylog.plugins.views.search.permissions.SearchUser;
+import org.graylog2.cluster.NodeService;
+import org.graylog2.plugin.rest.PluginRestResource;
+import org.graylog2.rest.RemoteInterfaceProvider;
+import org.graylog2.rest.RestTools;
+import org.graylog2.shared.rest.resources.ProxiedResource;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+
+import static org.graylog.plugins.views.search.rest.SearchResource.SEARCH_FORMAT_V1;
+import static org.graylog2.shared.rest.documentation.generator.Generator.CLOUD_VISIBLE;
+
+@Api(value = "SearchJobs", tags = {CLOUD_VISIBLE})
+@Path("/views/searchjobs")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes({MediaType.APPLICATION_JSON})
+@RequiresAuthentication
+public class SearchJobsStatusResource extends ProxiedResource implements PluginRestResource {
+
+    @Inject
+    public SearchJobsStatusResource(@Context HttpHeaders httpHeaders,
+                                    NodeService nodeService,
+                                    RemoteInterfaceProvider remoteInterfaceProvider,
+                                    ExecutorService executorService) {
+        super(httpHeaders, nodeService, remoteInterfaceProvider, executorService);
+    }
+
+    @GET
+    @ApiOperation(value = "Retrieve the status of an executed query")
+    @Path("{nodeId}/{jobId}/status")
+    @Produces({MediaType.APPLICATION_JSON, SEARCH_FORMAT_V1})
+    public void asyncSearchJobStatus(@ApiParam(name = "jobId") @PathParam("jobId") String jobId,
+                                     @ApiParam(name = "nodeId") @PathParam("nodeId") String nodeId,
+                                     @Context SearchUser searchUser,
+                                     @Suspended AsyncResponse asyncResponse) {
+        processAsync(asyncResponse,
+                () -> {
+                    try {
+                        final NodeResponse<ResponseBody> nodeResponse = requestOnNode(nodeId, r -> r.jobStatus(jobId), RemoteSearchJobsStatusInterface.class);
+                        return RestTools.streamResponse(nodeResponse, MediaType.APPLICATION_JSON, null);
+                    } catch (IOException e) {
+                        return Response.serverError().entity(e.getMessage()).build();
+                    }
```

Comment:
What's the reason for streaming the JSON response here instead of just returning it?

Reply:
Is there any disadvantage of using it?

Reply:
Response can be thousands of lines, hundreds of KB.
So I thought it is a similar case. 
I am not aware of any disadvantages...

Reply:
Alright.